packages/cli-kit/src/private/node/ui/components/Table/Row.tsx

```diff
@@ -0,0 +1,55 @@
+import ScalarDict from './ScalarDict.js'
+import {Column} from './Column.js'
+import {Box, Text} from 'ink'
+import React from 'react'
+
+interface RowProps<T extends ScalarDict> {
+  line: string
+  rowKey: string
+  data: Partial<T>
+  columns: Column<T>[]
+}
+
+function intersperse<T, TI>(intersperser: (index: number) => TI, elements: T[]): (T | TI)[] {
+  return elements.reduce((elements, element, index) => {
+    if (elements.length === 0) {
+      return [element]
+    }
+    return [...elements, intersperser(index), element]
+  }, [] as (T | TI)[])
+}
+
+export function Row<T extends ScalarDict>(props: RowProps<T>) {
+  const {line} = props
+
+  return (
+    <Box flexDirection="row">
+      {...intersperse(
+        (index) => {
+          const key = `${props.rowKey}-horizontal-separator-${index}`
+          return <Text key={key}>{'  '}</Text>
+        },
+
+        props.columns.map((column) => {
+          const content = props.data[column.name]
+
+          if (content === undefined || content === null) {
+            const key = `${props.rowKey}-empty-${column.name.toString()}`
```

Comment:
Do we need a separate key? I'm thinking that maybe we could remove this early return and apply the same logic to all rows. In this case because `String(content).length` is `0`, then `const marginRight = column.width - String(content).length` would be equal to `column.width` which is what this branch covers. If we removed this branch everything should still be working.

If it doesn't let you use `undefined` below you can always do `{content ?? ''}`

Reply:
Agreed, I will change it :)